runtime/compiler/runtime/SignalHandler.c

```diff
@@ -31,6 +32,31 @@
 
 #if defined(J9VM_PORT_SIGNAL_SUPPORT) && defined(J9VM_INTERP_NATIVE_SUPPORT)
 
+#if defined(TR_HOST_X86) && defined(TR_TARGET_X86)
+static UDATA isDfSet(J9VMThread* vmThread, void *sigInfo)
+{
+	PORT_ACCESS_FROM_VMC(vmThread);
+	const char* infoName;
+	void* infoValue;
+	U_32 infoType;
+	UDATA eflags;
+	const char* enableDFCheck = getenv("TR_trapIfDFSet");
+
+	if (enableDFCheck) {
+		infoType = j9sig_info(sigInfo, J9PORT_SIG_CONTROL, J9PORT_SIG_CONTROL_X86_EFLAGS, &infoName, &infoValue);
+		if (infoType == J9PORT_SIG_VALUE_ADDRESS) {
+			eflags = *(UDATA*)infoValue;
+			if ((eflags & 0x400) != 0) {
+				fprintf(stdout, "eflags %lx, eflags & 0x400 = %lx, DF flag is set\n", eflags, eflags & 0x400);
```

Comment:
As a rule, the VM prints to stderr. If you use `j9tty_printf` you'll get this for free. UDATA-sized values are `%zx` in the port library printf.